external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -53,15 +53,11 @@ import org.apache.spark.sql.types._
 import org.apache.spark.sql.v2.avro.AvroScan
 import org.apache.spark.util.Utils
 
-abstract class AvroSuite
+abstract class AvroSuiteBase
```

Comment:
why do we need to touch this suite? The fix is for catalog stuff and is not related to the file source.

Reply:
After running the test in `CommonFileDataSourceSuite`, `org.apache.hadoop.fs.Path.getFileSystem()` in other tests in the same suite will fail the check in `FakeFileSystemRequiringDSOption`, since they share the same `SparkContext`. 

That's why I split the suites that extends `CommonFileDataSourceSuite` into separate ones in this change.

Reply:
is it because `org.apache.hadoop.fs.Path.getFileSystem()` has cache?

Reply:
It's because `SparkContext.hadoopConfiguration` is passed to `org.apache.hadoop.fs.Path.getFileSystem()` and that is shared across tests in the same suite.

Reply:
but certain test case can reset the hadoop configs it sets to avoid polluting the shared environment, right?

Reply:
The test in `CommonFileDataSourceSuite` actually uses `withSQLConf` which does reset the config. However `SessionCatalog.hadoopConf` are set and could not be changed unless we create a new `SessionCatalog`